Book entity, repository and service layers add

jszymanski97 opened the pull request:
Book entity created
BookRepository and impl created
BookService and impl created
application.properties modified to connect db
BookStoreApp functions tested

jszymanski97 committed:
Book entity created
BookRepository and impl created
BookService and impl created
application.properties modified to connect db
BookStoreApp functions tested

beatarjn commented on src/main/java/mate/project/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,55 @@
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
```
Pls remove empty line

beatarjn commented on src/main/java/mate/project/OnlineBookStoreAppApplication.java:
```diff
@@ -1,13 +1,40 @@
+        return new CommandLineRunner() {
+            @Override
+            public void run(String... args) throws Exception {
+                Book book = new Book();
```
Book creation could be extracted to a separate private method for better readability

beatarjn commented on src/main/java/mate/project/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+
+@Service
+public class BookServiceImpl implements BookService {
+
```
Pls don't start class with an empty line

beatarjn commented on src/main/java/mate/project/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+        return bookRepository.findAll();
+    }
+
+    @Autowired
```
Constructor injection is generally preferred because it enforces the dependency requirement at the time of object creation and it's easier to test. You can use also @AllArgsConstructor annotation

jszymanski97 committed:
OnlineBookStoreAppApplication - new createBook method
BookServiceImpl - changed injection type to use constructor
BookRepositoryImpl - redundant empty line removed

MateuszRostek commented on src/main/java/mate/project/OnlineBookStoreAppApplication.java:
```diff
@@ -25,7 +25,7 @@ CommandLineRunner commandLineRunner() {
             public void run(String... args) throws Exception {
-                Book book = new Book();
+                Book book = createBook();
                 book.setTitle("Book Title");
                 book.setAuthor("Author");
                 book.setIsbn("isbn");
```
It should all be in the `createBook` method.

MateuszRostek reviewed (commented):
I've approved this Pull Requesst last week, you should only make one change that I've suggested and then you should merge this branch to the`main`. Then you should create a new branch and provide solution for the next task there. At the end, you should create another Pull Request (do not push changes into the same one).

jszymanski97 commented:
@MateuszRostek  right... I see I din't quite get what to do with the next task but now it makes sense... I will revert my previous commit and push the changes to the main. Than crerate new pull request with the new task and also the suggested change from previus one
